From 9217e7ff0df351af4087b2e8fa1ed7ce24888b7c Mon Sep 17 00:00:00 2001
From: Anson Huang <b20788@freescale.com>
Date: Mon, 24 Mar 2014 13:37:36 +0800
Subject: [PATCH 442/449] ENGR00304841-2 linux-test: wdog: add dbg monitor
 support

As debug monitor need to work with wdog, so put it in
the wdog folder. To use debug monitor, follow below steps:

1. Run dbg_monitor.sh at background;
2. Run your test case;
3. When system hang, wdog will trigger system reboot
   in 5 seconds, then the last AXI address info will
   be output to uboot, such as below:

U-Boot 2013.04-00158-gd17f24d-dirty (Mar 24 2014 - 11:37:58)

CPU:   Freescale i.MX6SX rev1.0 at 792 MHz
CPU:   Temperature 38 C, calibration data: 0x58c00000
DBG snvs regs(addr, data and info): 0x8f8010, 0x179c1183, 0x70000
Reset cause: unknown reset
Board: MX6SX SABRE SDB
I2C:   ready
DRAM:  1 GiB
MMC:   FSL_SDHC: 0, FSL_SDHC: 1, FSL_SDHC: 2
In:    serial
Out:   serial
Err:   serial
Found PFUZE100! deviceid 0x10, revid 0x21
mmc2 is current device
Net:   FEC0
Normal Boot
Hit any key to stop autoboot:  0
=>

Signed-off-by: Anson Huang <b20788@freescale.com>
---
 test/wdog/Makefile       |  3 ++-
 test/wdog/dbg_monitor.sh | 19 +++++++++++++++++++
 2 files changed, 21 insertions(+), 1 deletion(-)
 create mode 100755 test/wdog/dbg_monitor.sh

diff --git a/test/wdog/Makefile b/test/wdog/Makefile
index 6a629cb..8e400e5 100644
--- a/test/wdog/Makefile
+++ b/test/wdog/Makefile
@@ -4,7 +4,8 @@ EXCLUDE_LIST:=
 ifeq (,$(findstring $(PLATFORM), $(EXCLUDE_LIST)))
 OBJS = \
 	$(OBJDIR)/wdt_driver_test.out \
-	$(OBJDIR)/autorun-wdog.sh
+	$(OBJDIR)/autorun-wdog.sh \
+	$(OBJDIR)/dbg_monitor.sh
 
 else
 OBJS =
diff --git a/test/wdog/dbg_monitor.sh b/test/wdog/dbg_monitor.sh
new file mode 100755
index 0000000..794796a
--- /dev/null
+++ b/test/wdog/dbg_monitor.sh
@@ -0,0 +1,19 @@
+#!/bin/bash
+# demo for dbg_monitor module
+
+# reset and ungate dbg monitor;
+/unit_tests/memtool 0x21cc000=0;
+
+# monitor all address range;
+/unit_tests/memtool 0x21cc000=0x1;
+
+# enable wdog reset;
+/unit_tests/wdt_driver_test.out 5 1 0 &
+
+# enable wdog irq to trigger dbg monitor;
+/unit_tests/memtool -16 0x20bc006=0x8001;
+
+# after system hang and wdog auto reset, read address
+# of 0x21cc060, 0x21cc070, 0x21cc080 to get the last
+# axi address, data, master ID, and status.
+
-- 
1.8.0

